ui/src/routes/_layout/organizations/$orgId/products/$productId/-components/product-packages-statistics-card.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { Boxes } from 'lucide-react';
+
+import { Package } from '@/api/requests';
+import { StatisticsCard } from '@/components/statistics-card';
+import { getEcosystemBackgroundColor } from '@/helpers/get-status-class';
+import { usePackagesByProductIdSuspense } from '@/hooks/use-packages-by-product-suspense';
+import { cn } from '@/lib/utils';
+import { PackageManager } from '../repositories/$repoId/-types';
+
+type ProductPackagesStatisticsCardProps = {
+  productId: number;
+  className?: string;
+};
+
+/**
+ * Calculate the counts of packages by their ecosystem.
+ *
+ * @param packages
+ * @returns Package counts sorted by ecosystem.
+ */
+const calcPackageEcosystemCounts = (
+  packages: Package[]
+): { ecosystem: PackageManager; count: number }[] => {
+  const ecosystems = packages.map(
+    (pkg) => pkg.identifier.type as PackageManager
+  );
+  const counts = ecosystems.reduce(
+    (acc, ecosystem) => {
+      acc[ecosystem] = (acc[ecosystem] || 0) + 1;
+      return acc;
+    },
+    {} as Record<PackageManager, number>
+  );
+
+  // Sort the counts alphabetically by ecosystem.
+  return Object.entries(counts)
+    .sort(([a], [b]) => a.localeCompare(b))
+    .map(([ecosystem, count]) => ({
+      ecosystem: ecosystem as PackageManager,
+      count,
+    }));
+};
+
+export const ProductPackagesStatisticsCard = ({
+  productId,
+  className,
+}: ProductPackagesStatisticsCardProps) => {
+  const data = usePackagesByProductIdSuspense({
+    productId: productId,
+  });
+
+  const packagesTotal = data?.length;
+
+  return (
+    <StatisticsCard
+      title='Packages'
+      icon={() => <Boxes className='h-4 w-4 text-green-500' />}
+      value={packagesTotal}
+      counts={
+        packagesTotal
+          ? calcPackageEcosystemCounts(data).map(({ ecosystem, count }) => ({
+              key: ecosystem,
+              count,
+              color: getEcosystemBackgroundColor(ecosystem),
```

Comment:
And here I would get the ecosystemBackgroundColor if the string is an ecosystem and some default color otherwise.